Tests/libhostmgrTests/resources/dotenv-fixtures.env

```diff
@@ -0,0 +1,26 @@
+UNQUOTED_STRING=foo
+QUOTED_STRING="foo"
+
+UNQUOTED_STRING_WITH_LEADING_SPACE_IN_VALUE= foo
+UNQUOTED_STRING_WITH_TRAILING_SPACE_IN_VALUE=foo 
+
+STRING_WITH_QUOTED_LEADING_SPACE=" foo"
+STRING_WITH_QUOTED_TRAILING_SPACE="foo "
+
+STRING_WITH_LEADING_QUOTE=""foo"
```

Comment:
I don't think that's valid syntax for a regular dotenv file. i.e. that would lead to a syntax error if parsed with the previous library we used or with `source` from a shell. Yet, your parser interprets it in its own way and differently to the spec and to how `source` would.